### PR TITLE
Add support for CodeCommit repositories

### PR DESCRIPTION
**Pull request by reedloden**

### CODE REVIEW

**Comment by reedloden** [comment]
@itmecho This one is rebased and ready for review.

File: internal/awsdata/codecommit.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		if out.NextToken == nil {
+			done = true
+		} else {
+			params.NextToken = out.NextToken
+		}
+	}
+
+	log.Info("processing data")
+
+	// TODO: API call can only handle 100 repository names at a time
```
Comment (itmecho) [inline internal/awsdata/codecommit.go:61]: I think this is fine as a starting point

File: internal/awsdata/codecommit.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	out, err := codecommitSvc.BatchGetRepositories(&codecommit.BatchGetRepositoriesInput{
```
Comment (itmecho) [inline internal/awsdata/codecommit.go:62]: I think we should check if the repo list is empty before this step and just return early. Currently if you have no code commit repos in your account, you get the following error:
```
ERRO[0000] RepositoryNamesRequiredException: repositoryNames cannot be empty
```

Comment (itmecho) [reply internal/awsdata/codecommit.go:62]: This might be a problem in other loaders as well but I haven't tested yet

**Review by itmecho — CHANGES_REQUESTED** [review]

File: internal/awsdata/codecommit.go (hunk 2 above)
Comment (reedloden) [reply internal/awsdata/codecommit.go:62]: Good catch! Fixed here and other places. Just went ahead and included those fixes in this PR.

**Review by itmecho — APPROVED** [review]

**Comment by itmecho** [comment]
Thanks for the PR!

**Pull request merged**